snooty/test_parser.py

```diff
@@ -132,6 +132,35 @@ def test_chapter() -> None:
     assert isinstance(diagnostics[0], CannotOpenFile)
 
 
+def test_product() -> None:
+    """Test product directive"""
+    path = FileId("test.rst")
+    project_config = ProjectConfig(ROOT_PATH, "", source="./")
+    parser = rstparser.Parser(project_config, JSONVisitor)
+    PRODUCT_CONTENT = """
+.. landing:products::
+
+   .. landing:product:: Test Atlas
+      :icon: test_project/source/general-features-tools.svg
+      :icon-alt: Test Atlas
+
+      Test Product Description.
+
+      .. cta::
+      
+         This card was built with a relative URL.
+"""
+
+    page, diagnostics = parse_rst(
+        parser,
+        path,
+        PRODUCT_CONTENT,
+    )
+
+    page.finish(diagnostics)
```

Comment:
i would also validate the page.ast with `check_ast_testing_string`

Reply:
@seungpark I may need to circle back on this one. I have a bit more I need to learn and understand, and I don't think this should block merging. I will definitely work on this though.